templates/components/form/form.jsx

```diff
@@ -56,7 +75,8 @@ var Form = React.createClass({
     }
 
     return (
-     <label className={className} key={id} htmlFor={id}>
+     <label ref={id+'Label'} className={this.getInLabelClasses(id)} key={id} htmlFor={id}>
+        {!this.isValid(id) ? errorTooltip : ''}
```

Comment:
we're calling `this.isValid(id)` a lot, even though we have a `dirty` state that tells us whether or not things are valid, right? There should really only ever be one call to `this.isValid()`, everything else we should be able to get from `this.state....`